src/model/SignedIntent.ts

```diff
@@ -42,23 +43,85 @@ export class SignedIntent {
 
     relay(provider: Provider) {
         if (provider === null || provider === undefined) {
-            throw Error("The provider can not be null or undefined");
             provider = Provider.getGlobal();
-            if (provider === null) throw Error("A valid configuration must be provided or set as global");
+            if (provider === null || provider === undefined) throw Error("A valid configuration must be provided or set as global");
         }
         if (provider.relayer === null || provider.relayer === undefined) {
             throw Error("The provider is invalid, have not relayer");
         }
 
         const relayClient = new RelayClient(provider.relayer);
-        relayClient.post(this);
+        relayClient.post(this).then(response => {
+            if (response.statusCode !== 200) throw Error("The provider is invalid, have not relayer");
+        })
     }
 
-    status(provider: Provider) {
+    async status(provider: Provider): Promise<Status> {
         if (provider === null || provider === undefined) {
-            throw Error("The provider can not be null or undefined");
+            throw Error("The provider can not be null or undefined")
         }
-        // TODO: CODE
+
+        const minConfirmation = 32;
+        const web3 = provider.web3
+        const signatureRelayedEvent = web3.eth.abi.encodeEventSignature('Relayed(bytes32,bytes32[],address,uint256,bytes,bytes32,uint256,bool)')
+        const currentBlockNumber = await web3.eth.getBlockNumber()
+        const data = web3.eth.abi.encodeFunctionCall({
```

Comment:
Consider using more descriptive names for those `data` fields, maybe relayedData or something similar

Reply:
done